sycl/test-e2e/Basic/library_loading.cpp

```diff
@@ -11,9 +13,9 @@
 using namespace sycl;
 
 int main() {
-  // CHECK-NO-FILTER: {{(SYCL_PI_TRACE\[-1\]: dlopen\(.*/libpi_cuda.so\) failed with)|(SYCL_PI_TRACE\[basic\]: Plugin found and successfully loaded: libpi_cuda.so)}}
+  // CHECK-NO-FILTER-CUDA: {{(SYCL_PI_TRACE\[-1\]: dlopen\(.*/libpi_cuda.so\) failed with)|(SYCL_PI_TRACE\[basic\]: Plugin found and successfully loaded: libpi_cuda.so)}}
   // CHECK-NO-FILTER-NOT: {{(SYCL_PI_TRACE\[-1\]: dlopen\(.*/libpi_esimd_emulator.so\) failed with)|(SYCL_PI_TRACE\[basic\]: Plugin found and successfully loaded: libpi_esimd_emulator.so)}}
-  // CHECK-NO-FILTER: {{(SYCL_PI_TRACE\[-1\]: dlopen\(.*/libpi_hip.so\) failed with)|(SYCL_PI_TRACE\[basic\]: Plugin found and successfully loaded: libpi_hip.so)}}
+  // CHECK-NO-FILTER-HIP: {{(SYCL_PI_TRACE\[-1\]: dlopen\(.*/libpi_hip.so\) failed with)|(SYCL_PI_TRACE\[basic\]: Plugin found and successfully loaded: libpi_hip.so)}}
```

Comment:
Would it not be enough to just use `CHECK-NO-FILTER-DAG`?

Reply:
@steffenlarsen @cperkinsintel I am using `CHECK-NO-FILTER-DAG` now.

The test now is reduced to 2 runs of `FileCheck` for the original `CHECK-NO-FILTER`.  Mixing `CHECK-NO-FILTER-NOT` with `CHECK-NO-FILTER-DAG` would not give the result we want to achieve with the test. In case we put `CHECK-NO-FILTER-NOT` between the two `CHECK-NO-FILTER-DAG`, there will be no reordering. In the other cases, FileCheck won't match the regex with entire file.

see https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-dag-directive